contain-intrinsic-* missing properties (#21110)
mdn/content: files/en-us/web/css/contain-intrinsic-size/index.md

---
title: contain-intrinsic-size
slug: Web/CSS/contain-intrinsic-size
tags:
  - CSS
  - CSS Property
  - Reference
  - experimental
  - contain-intrinsic-size
  - recipe:css-shorthand-property
page-type: css-property
browser-compat: css.properties.contain-intrinsic-size
---

{{CSSRef}} {{SeeCompatTable}}

> The **`contain-intrinsic-size`** [CSS](/en-US/docs/Web/CSS) [shorthand property](/en-US/docs/Web/CSS/Shorthand_properties) sets the size of an element that will be used for layout when it is subject to [size containment](/en-US/docs/Web/CSS/CSS_Containment#size_containment).

Reviewer: ```suggestion
The **`contain-intrinsic-size`** [CSS](/en-US/docs/Web/CSS) [shorthand property](/en-US/docs/Web/CSS/Shorthand_properties) sets the size of an element that a browser will use for layout when the element is subject to [size containment](/en-US/docs/Web/CSS/CSS_Containment#size_containment).
```

{{csssyntax}}

## Examples

> ### Setting the intrinsic size

Author: @dipikabh With respect to your comment above about missing Examples, of course you are right that they are desirable. I was hoping to get this in and let someone else who is better with CSS do a better example :-)

That said, I have tried just this one case and if it is heading in the right track for a minimal example then I'll expand it a bit. What do you think of the example below (partial - mostly after "does it seem helpful").

What this does is allow a user to toggle the visibility and contains-intrinsic-size of an element that has content. What it can be extended to show (i.e. when I update the words) is how the intrinsic size collapses to nothing if it is not set, however this does not apply if the item is visible unless it is in size containment. Still a little bit of work to go.

Reviewer: Sounds good to me. I can very much relate to "let someone else who is better with CSS do a better example" :)
